### PR TITLE
Fix/users middleware

### PR DESCRIPTION
**Pull request by y-golde**
Added 2 new middlewares 

### Users middleware
handles any request regarding users ( where a user preforms the action on another user ) using the following logic : 
if the user is a super-admin: allow the request if the requested user is in the asking user's district
if the user is an admin: allow the request if the requested user is in the asking user's county
if the user is a normal investigator:  deny the request

### counties middleware
handles requests regarding counties (see counties users/investigations) using the following logic : 
if the user is a super-admin: allow the request if the county is in the asking user's district
if the user is an admin: allow the request if the county is the asking user's county
if the user is a normal investigator: deny the request

added an alert for errored requests in adminLandingPage

### CODE REVIEW

File: server/src/middlewares/HandleCountyRequest.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { NextFunction, Request, Response } from 'express';
+
+import UserType from '../Models/User/UserType';
+import { DISTRICT_BY_COUNTY } from '../DBService/Counties/Query';
```
Comment (Noam6421) [inline server/src/middlewares/HandleCountyRequest.ts:4]: fix imports order

File: server/src/middlewares/HandleCountyRequest.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const handleCountyRequest = async (request: Request, response: Response, next: NextFunction) => {
+    const currentUser = response.locals.user;
+    const county = parseInt(request.body.county);
+
+    const countyMiddlewareLogger = logger.setup({
+        workflow: 'countyMiddleware',
+        investigation: currentUser.epidemiologynumber,
+    });
+
+    if (currentUser.userType === UserType.SUPER_ADMIN) {
```
Comment (Noam6421) [inline server/src/middlewares/HandleCountyRequest.ts:18]: maybe export 'currentUser.userType' to const (line 33 also)

File: server/src/middlewares/HandleUsersRequest.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { NextFunction, Request, Response } from 'express';
+
+import UserType from '../Models/User/UserType';
+import { DISTRICT_COUNTY_BY_USER } from '../DBService/Users/Query';
```
Comment (Noam6421) [inline server/src/middlewares/HandleUsersRequest.ts:4]: also fix imports order

File: server/src/middlewares/HandleUsersRequest.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    const currentUser = response.locals.user;
+    const { userId } = request.body;
+
+    const usersMiddlewareLogger = logger.setup({
+        workflow: 'InvestigationMiddleware',
+        investigation: currentUser.epidemiologynumber,
+    });
+
+    const questionedUser = await getUserDistrictCounty(userId, response.locals);
+    if (currentUser.userType === UserType.SUPER_ADMIN) {
```
Comment (Noam6421) [inline server/src/middlewares/HandleUsersRequest.ts:19]: 'currentUser.userType' maybe move to const?

**Review by Noam6421 — CHANGES_REQUESTED** [review]

File: server/src/middlewares/HandleCountyRequest.ts (hunk 1 above)
Comment (y-golde) [reply server/src/middlewares/HandleCountyRequest.ts:5]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1249/commits/51a1dbb29c72a734e562a38a45e089f0bea2498b)

File: server/src/middlewares/HandleCountyRequest.ts (hunk 2 above)
Comment (y-golde) [reply server/src/middlewares/HandleCountyRequest.ts]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1249/commits/51a1dbb29c72a734e562a38a45e089f0bea2498b)

File: server/src/middlewares/HandleUsersRequest.ts (hunk 3 above)
Comment (y-golde) [reply server/src/middlewares/HandleUsersRequest.ts:5]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1249/commits/51a1dbb29c72a734e562a38a45e089f0bea2498b)

File: server/src/middlewares/HandleUsersRequest.ts (hunk 4 above)
Comment (y-golde) [reply server/src/middlewares/HandleUsersRequest.ts]: fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1249/commits/51a1dbb29c72a734e562a38a45e089f0bea2498b)

**Review by Noam6421 — APPROVED** [review]

**Review by chenFriedman — APPROVED** [review]

**Comment by MagusM** [comment]
@y-golde 
Please leave this PR open. some changes needs to be done due to change of task definition.

**Comment by MagusM** [comment]
tasks specifications needs to be reviewed little changes needs to be made accordingly,  a new MR will be opened once done.

**Pull request merged**